numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1005,14 +1006,24 @@ arr_ravel_multi_index(PyObject *self, PyObject *args, PyObject *kwds)
             s = 1;
             for (i = dimensions.len-1; i >= 0; --i) {
                 ravel_strides[i] = s;
-                s *= dimensions.ptr[i];
+                if (npy_mul_with_overflow_intp(&s, s, dimensions.ptr[i])) {
+                    PyErr_SetString(PyExc_ValueError,
+                    "invalid dims: array size defined by dims is larger "
```

Comment:
Thanks, will merge soon unless someone else wants to chime in I think. Remind me if I forget ;).